x/logic/keeper/interpreter.go

```diff
@@ -3,13 +3,15 @@ package keeper
 import (
 	goctx "context"
 	"math"
+	"strings"
 
 	"github.com/ichiban/prolog"
+	"github.com/ichiban/prolog/engine"
 	"github.com/samber/lo"
+	"golang.org/x/net/context"
```

Comment:
Nope. `context` and `golang.org/x/net/context` are different packages.

Reply:
Yes, absolutely, indeed. I suspect an auto-import might have inadvertently introduced the golang.org/x/net/context package. However, upon initial review, I am fully capable of removing this package by restructuring the code. This will indeed make it cleaner. Let it be done!